src/main/java/ch/njol/skript/classes/data/DefaultOperations.java

```diff
@@ -0,0 +1,124 @@
+		});
+		Arithmetics.registerDefaultValue(Number.class, () -> 0L);
+
+		// Vector - Vector
+		Arithmetics.registerOperation(Operator.ADDITION, Vector.class, (left, right) -> left.clone().add(right));
+		Arithmetics.registerOperation(Operator.SUBTRACTION, Vector.class, (left, right) -> left.clone().subtract(right));
+		Arithmetics.registerOperation(Operator.MULTIPLICATION, Vector.class, (left, right) -> left.clone().multiply(right));
+		Arithmetics.registerOperation(Operator.DIVISION, Vector.class, (left, right) -> left.clone().divide(right));
+		Arithmetics.registerDifference(Vector.class,
+			(left, right) -> new Vector(Math.abs(left.getX() - right.getX()), Math.abs(left.getY() - right.getY()), Math.abs(left.getZ() - right.getZ())));
+		Arithmetics.registerDefaultValue(Vector.class, Vector::new);
+
+		// Vector - Number
+		// Number - Vector
+		Arithmetics.registerOperation(Operator.MULTIPLICATION, Vector.class, Number.class, (left, right) -> left.clone().multiply(right.doubleValue()), (left, right) -> {
+			double number = left.doubleValue();
+			Vector leftVector = new Vector(number, number, number);
+			return leftVector.multiply(right);
+		});
+		Arithmetics.registerOperation(Operator.DIVISION, Vector.class, Number.class, (left, right) -> {
+			double number = right.doubleValue();
+			Vector rightVector = new Vector(number, number, number);
+			return left.clone().divide(rightVector);
+		}, (left, right) -> {
+			double number = left.doubleValue();
+			Vector leftVector = new Vector(number, number, number);
+			return leftVector.divide(right);
+		});
+
+		// Timespan - Timespan
+		Arithmetics.registerOperation(Operator.ADDITION, Timespan.class, (left, right) -> new Timespan(left.getMilliSeconds() + right.getMilliSeconds()));
+		Arithmetics.registerOperation(Operator.SUBTRACTION, Timespan.class, (left, right) -> new Timespan(Math.max(0, left.getMilliSeconds() - right.getMilliSeconds())));
+		Arithmetics.registerDifference(Timespan.class, (left, right) -> new Timespan(Math.abs(left.getMilliSeconds() - right.getMilliSeconds())));
+		Arithmetics.registerDefaultValue(Timespan.class, Timespan::new);
+
+		// Timespan - Number
+		// Number - Timespan
+		Arithmetics.registerOperation(Operator.MULTIPLICATION, Timespan.class, Number.class, (left, right) -> {
+			long scalar = right.longValue();
+			if (scalar < 0)
+				return null;
+			return new Timespan(left.getMilliSeconds() * scalar);
+		}, (left, right) -> {
+			long scalar = left.longValue();
+			if (scalar < 0)
+				return null;
+			return new Timespan(scalar * right.getMilliSeconds());
+		});
+		Arithmetics.registerOperation(Operator.DIVISION, Timespan.class, Number.class, (left, right) -> {
+			long scalar = right.longValue();
+			if (scalar <= 0)
+				return null;
+			return new Timespan(left.getMilliSeconds() / scalar);
+		}, (left, right) -> {
+			long scalar = left.longValue();
+			if (scalar < 0)
+				return null;
+			return new Timespan(scalar / right.getMilliSeconds());
+		});
```

Comment:
I can see why you made division commutative too, but I don't think it should be.
1 / (1 second) should really be 1 second^-1, not 1 second, so i think number / timespan should be invalid.

Reply:
> 1 / (1 second) should really be 1 second^-1

So you're saying that I should add an exponentiation operation for timespan - number?

Reply:
no no
second^-1 is notation for "per second"
i'm just saying you shouldn't be able to divide a number by a timespan